libglusterfs/src/ctx.c

```diff
@@ -57,6 +57,12 @@ glusterfs_ctx_new()
     GF_ATOMIC_INIT(ctx->stats.total_pairs_used, 0);
     GF_ATOMIC_INIT(ctx->stats.total_dicts_used, 0);
 
+    if (gethostname(ctx->hostname, HOST_NAME_MAX + 1)) {
```

Comment:
I think buffer name should be LOGIN_NAME_MAX, there is no macro exist as a HOST_NAME_MAX

Reply:
IIUC fully portable and POSIX-conformant solution is `sysconf(_SC_HOST_NAME_MAX)` and fallback to `_POSIX_HOST_NAME_MAX` in case of error.